frontend/js/src/common/brainzplayer/BrainzPlayer.tsx

```diff
@@ -945,6 +951,75 @@ export default function BrainzPlayer() {
     }
   };
 
+  const handleAlbumMapping = (
+    dataSource: keyof MatchedTrack,
+    releaseMBID: string,
+    album: {
+      trackName: string;
+      uri: string;
+    }[]
+  ): void => {
+    const currentQueue = queueRef.current;
+    const currentAmbientQueue = ambientQueueRef.current;
+
+    // Filter the currentQueue after the current playing track
+    const currentPlayingListenIndex = currentListenIndexRef.current;
+    const currentQueueAfterCurrent = currentQueue.slice(
+      currentPlayingListenIndex + 1
+    );
+
+    const newQueueMatchedTracks: Record<string, string> = {};
+    const newAmbientQueueMatchedTracks: Record<string, string> = {};
+
+    // Filter the tracks in the album that are not yet matched
+    const queueTracksInAlbum = currentQueueAfterCurrent.filter((track) => {
+      return (
+        getReleaseMBID(track) === releaseMBID &&
```

Comment:
@MonkeyDo Should we exclude tracks without a releaseMBID? In some cases, a listen might lack this mapping data but still have the same trackName we’re searching for.

Reply:
I'm reading all this for the first time, so i hope I won't say anything too stupid XD

I don't think we should filter out non-mapped tracks, but I also don't understand why you want to do that in the first place.
Considering the mapping is a best-effort sort of feature, assume it won't always be present.

You can additionally use the `releaseName` and do some fuzzy matching on that if you are trying to group all tracks from the same album.